frontend/src/scenes/surveys/utils/branchingPreviewLogic.ts

```diff
@@ -0,0 +1,79 @@
+import {
+    BasicSurveyQuestion,
+    LinkSurveyQuestion,
+    MultipleSurveyQuestion,
+    RatingSurveyQuestion,
+    SurveyQuestionBranchingType,
+    SurveyQuestionType,
+} from '~/types'
+
+export function getRatingScaleResponse(response: number, scale: number): string {
+    switch (scale) {
+        case 10: // NPS scale
+            if (response <= 6) {
+                return 'detractors'
+            }
+            if (response <= 8) {
+                return 'passives'
+            }
+            return 'promoters'
+        case 7: // 7-point likert scale
+            if (response <= 3) {
+                return 'negative'
+            }
+            if (response === 4) {
+                return 'neutral'
+            }
+            return 'positive'
+        case 5: // 5-point scale
+            if (response <= 2) {
+                return 'negative'
+            }
+            if (response === 3) {
+                return 'neutral'
+            }
+            return 'positive'
+        default:
+            return 'neutral'
+    }
+}
+
+export function getNextQuestionIndex(
```

Comment:
not sure if this is a copy-pasta from the [JS SDK](https://github.com/PostHog/posthog-js/blob/4c50987442b6fb662792365604e524961ce00dd8/src/posthog-surveys.ts#L232), but if not, you could inspire from in case something isn't working well.